test/test.js

```diff
@@ -728,3 +729,75 @@ describe('pipeline execution', () => {
     });
   }
 });
+
+describe('function execution', () => {
+  for (const envConfig of envConfigs) {
+    describe(envConfig.name, () => {
+      beforeAll(envConfig.setup);
+      afterAll(envConfig.teardown);
+      describe(envConfig.name + ' extended json decode (incoming)', () => {
+        beforeAll(() => {
+          fetchMock.restore();
+          fetchMock.post(LOCALAUTH_URL, {userId: '5899445b275d3ebe8f2ab8a6'});
+          fetchMock.post(FUNCTION_URL, (name, arg1, arg2) => {
+            return JSON.stringify({result: [{x: {'$oid': hexStr}}]});
+          });
+        });
+
+        it('should decode extended json from function responses', () => {
+          expect.assertions(1);
+          let testClient = new StitchClient('testapp');
+          return testClient.login('user', 'password')
+            .then(() => testClient.executeFunction('testfunc', {items: [{x: {'oid': hexStr}}]}, 'hello'))
+            .then((response) => expect(response.result[0].x).toEqual(new ejson.bson.ObjectID(hexStr)));
+        });
+      });
+
+      describe('extended json encode (outgoing)', () => {
+        let requestArg;
+        beforeAll(() => {
+          fetchMock.restore();
+          fetchMock.post(LOCALAUTH_URL, {userId: hexStr});
+          fetchMock.post(FUNCTION_URL, (name, arg1, arg2) => {
+            requestArg = arg1;
+            return {result: [{x: {'$oid': hexStr}}]};
+          });
+        });
+
+        it('should encode objects to extended json for outgoing function request body', () => {
+          expect.assertions(1);
+          let requestBodyExtJSON = {name: 'testfunc', arguments: [{x: {'$oid': hexStr}}, 'hello']};
```

Comment:
nit: i think you could move this directly into an arg to `toEqual` on L773, since we don't actually use it anywhere else so its not necessary to assign it with `let`.